lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -308,56 +308,51 @@ public static Domain getDomain(Type type, DictionaryDescriptor dictionaryDescrip
 
         int dictionarySize = dictionaryPage.get().getDictionarySize();
         if (type.equals(BIGINT) && columnDescriptor.getPrimitiveType().getPrimitiveTypeName() == PrimitiveTypeName.INT64) {
-            List<Domain> domains = new ArrayList<>();
+            List<Long> values = new ArrayList<>();
```

Comment:
If this is the bottleneck then maybe long[] will be faster.

Reply:
`ValueSet` constructor currently does accept primitive arrays.

however, i will at least preallocate the list to the required size, since it's well known.